R/get-algorithm.R

```diff
@@ -0,0 +1,16 @@
+#' Get the criteria algorithmic logic and convert to an R logic condition.
+#'
+#' @param criteria The name of the inclusion or exclusion criteria to use.
+#'
+#' @return A character string.
+#' @keywords internal
+#'
+#' @examples
+#' get_algorithm_logic("hba1c")
+get_algorithm_logic <- function(criteria) {
+  algorithm |>
+    dplyr::filter(.data$name == criteria) |>
+    dplyr::pull(.data$logic) |>
+    stringr::str_replace_all("AND", "&") |>
+    stringr::str_replace_all("OR", "|")
```

Comment:
Why do we not just use `&` and `|` in the csv instead of replacing them here? Is it for readability in the csv?

Reply:
Because those are more R logic symbols, not universal logic symbols: https://en.wikipedia.org/wiki/List_of_logic_symbols `&` is fine, but `|` is not a recognized math symbol for OR.

Reply:
Ah, I see! Thanks for elaborating.